panic: runtime error: invalid memory address or nil pointer dereference

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x10ae11f]

goroutine 1 [running]:
main.parseSmaliFiles.func1(0x7ffeefbff9ac, 0x13, 0x0, 0x0, 0x10fb740, 0xc000070210, 0x10, 0x10cc360)
	/Users/enlong/go/src/github.com/alexeysoshin/smali2java/smali2java.go:31 +0x2f
path/filepath.Walk(0x7ffeefbff9ac, 0x13, 0xc00004c2d0, 0x1, 0x0)
	/usr/local/Cellar/go/1.11.4/libexec/src/path/filepath/path.go:402 +0x6a
main.parseSmaliFiles(0x7ffeefbff9ac, 0x13)
	/Users/enlong/go/src/github.com/alexeysoshin/smali2java/smali2java.go:29 +0x89
main.main()
	/Users/enlong/go/src/github.com/alexeysoshin/smali2java/smali2java.go:23 +0x8d

```
➜  decompile smali2java
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x10b086f]

goroutine 1 [running]:
main.parseSmaliFiles.func1(0x10eab3b, 0x13, 0x0, 0x0, 0x1105ce0, 0xc000092060, 0x10, 0x10cf7a0)
	/usr/local/Cellar/go/1.12.5/libexec/src/github.com/AlexeySoshin/smali2java/smali2java.go:31 +0x2f
path/filepath.Walk(0x10eab3b, 0x13, 0xc0000900a0, 0x10ebc00, 0x0)
	/usr/local/Cellar/go/1.12.5/libexec/src/path/filepath/path.go:402 +0x6a
main.parseSmaliFiles(0x10eab3b, 0x13)
	/usr/local/Cellar/go/1.12.5/libexec/src/github.com/AlexeySoshin/smali2java/smali2java.go:29 +0x89
main.main()
	/usr/local/Cellar/go/1.12.5/libexec/src/github.com/AlexeySoshin/smali2java/smali2java.go:23 +0xe9
➜  decompile
```

--- a/smali2java.go
+++ b/smali2java.go
@@ -28,6 +28,10 @@ func parseSmaliFiles(path string) {
 	var wg = sync.WaitGroup{}
 	filepath.Walk(path, func(path string, f os.FileInfo, err error) error {
 
+		if err != nil {
+			log.Printf("Error occurred while walking path %q: %v", path, err)
+			return err
+		}
 		if f.IsDir() {
 			//log.Printf("Skipping directory: %s\n", path)
 		} else if filepath.Ext(path) != smaliExtension {

